pkg/postage/listener/listener.go

```diff
@@ -0,0 +1,264 @@
+package listener
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/postage"
+)
+
+const (
+	blockPage = 500 // how many blocks to sync every time
+	tailSize  = 100 // how many blocks to tail from the tip of the chain
```

Comment:
maybe let's decrease this to 20 for now, so that we can test things more easily

Reply:
done. although even 20 is quite long for testing.